### PR TITLE
Add the `cells_stubhead_label()` location helper

### PR DESCRIPTION
**Pull request by rich-iannone**
This adds the location helper function `cells_stubhead_label()`, and this makes it possible to add a footnote and styles to this single location.

Here is an example of how this works:

```r
library(gt)

sza %>%
  dplyr::filter(
    latitude == 20 &
      month == "jan" &
      !is.na(sza)
  ) %>%
  dplyr::select(-latitude, -month) %>%
  gt(rowname_col = "tst") %>%
  tab_header(title = "Solar Zenith Angles") %>%
  tab_stubhead_label(label = "tst") %>%
  tab_footnote(
    footnote = "Number indicates height of sun.",
    locations = cells_column_labels(
      columns = vars(sza))
  ) %>%
  tab_footnote(
    footnote = "True Solar Time.",
    locations = cells_stubhead_label()
  ) %>%
  tab_style(
    style = cell_fill(color = "lightgreen"),
    locations = cells_stubhead_label()
  ) %>%
  tab_style(
    style = cell_text(color = "purple"),
    locations = cells_stubhead_label()
  ) %>%
  tab_footnote(
    footnote = "Uses the `sza` dataset.",
    locations = cells_title(groups = "title")
  )
```

cells_stubhead_label

This is part of an effort to make all possible locations accessible for footnotes and styles (at this time, groupwise and grand summary captions in the stub still do not have location helpers). We eventually want to have all possible location helpers available to create a `cells_all()` location helper as recommended in https://github.com/rstudio/gt/issues/176.

Fixes: https://github.com/rstudio/gt/issues/224

### CODE REVIEW

File: R/tab_footnote.R [hunk 1]
```diff
@@ -86,37 +86,38 @@ set_footnote <- function(loc, data, footnote) {
...
-set_footnote.cells_data <- function(loc, data, footnote) {
+set_footnote.cells_title <- function(loc, data, footnote) {
 
-  resolved <- resolve_cells_data(data = data, object = loc)
+  if ((loc$groups %>% as.character())[-1] == "title") {
 
-  cols <- resolved$columns
-  rows <- resolved$rows
+    attr(data, "footnotes_df") <-
+      add_location_row(
```
Comment (schloerke) [inline R/tab_footnote.R]: `add_location_row_footnotes`

(and other method calls below)

Comment (rich-iannone) [reply R/tab_footnote.R]: Changed all the calls over to `add_location_row_footnotes()` in that file.

**Comment by rich-iannone** [comment]
Renamed `cells_stubhead_label()` to `cells_stubhead()`. Thanks for the good advice in https://github.com/rstudio/gt/issues/319!

**Comment by schloerke** [comment]
(Tests broke)

**Comment by rich-iannone** [comment]
Apologies for that. Now fixed (just had to do some renaming).

File: R/helpers.R [hunk 2]
```diff
@@ -244,6 +247,21 @@ cells_title <- function(groups = c("title", "subtitle")) {
   cells
 }
 
+#' @rdname location_cells
+#' @import rlang
+#' @export
+cells_stubhead <- function() {
+
+  # Create the `cells_stubhead` object
+  cells <- list(groups = "stubhead_label")
```
Comment (schloerke) [inline R/helpers.R]: `"stubhead"`

File: R/render_as_html.R [hunk 3]
```diff
@@ -21,6 +21,11 @@ render_as_html <- function(data) {
         footnotes_resolved, boxh_df, output = "html"
       )
 
+    # Add footnote glyphs to the stubhead label
+    stubhead_label <-
```
Comment (schloerke) [inline R/render_as_html.R]: stubhead

File: R/utils_render_footnotes.R [hunk 4]
```diff
@@ -452,6 +452,58 @@ set_footnote_glyphs_columns <- function(footnotes_resolved,
   boxh_df
 }
 
+#' Set footnote glyphs for the stubhead label
+#'
+#' @noRd
+set_footnote_glyphs_stubhead_label <- function(footnotes_resolved,
```
Comment (schloerke) [inline R/utils_render_footnotes.R]: stubhead

File: R/utils_render_html.R [hunk 5]
```diff
@@ -267,6 +282,11 @@ create_columns_component_h <- function(boxh_df,
     return("")
   }
 
+  # Get the style attrs for the stubhead label
+  stubhead_label_style_attrs <-
```
Comment (schloerke) [inline R/utils_render_html.R]: stubhead

File: tests/testthat/test-tab_footnote.R [hunk 6]
```diff
@@ -241,6 +241,28 @@ test_that("the `tab_footnote()` function works correctly", {
...
+  # a single row
+  attr(tab, "footnotes_df", exact = TRUE) %>%
+    nrow() %>%
+    expect_equal(1)
+
+  # Expect certain values for each of the columns in the
+  # single-row `footnotes_df` data frame
+  expect_attr_equal(
+    tab, "footnotes_df",
+    c("stubhead_label", "2.5", NA_character_, NA_character_, NA_character_,
```
Comment (schloerke) [inline tests/testthat/test-tab_footnote.R]: stubhead

File: R/utils_render_common.R [hunk 7]
```diff
@@ -703,13 +703,13 @@ process_heading <- function(heading, context) {
...
-process_stubhead_label <- function(caption, context) {
+# Process the `stubhead` object
+process_stubhead <- function(caption, context) {
 
   if (!is.null(caption)) {
-    stubhead_label <- caption$stubhead_label %>% process_text(context)
+    stubhead <- caption$stubhead %>% process_text(context)
 
-    return(list(stubhead_label = stubhead_label))
+    return(list(stubhead = stubhead))
```
Comment (schloerke) [inline R/utils_render_common.R]: I'm wondering if this is now actually the `stubhead_label` as it's the text of that `stubhead`

Comment (rich-iannone) [reply R/utils_render_common.R]: It is, but would it be okay to keep internal references to this always as `stubhead` and external references (e.g., `tab_stubhead_label()`, where we are defining the actual label text) as is?

**Pull request merged**